e2e_tests/tests/fixtures/mnist_pytorch/failable_model_def.py

```diff
@@ -0,0 +1,23 @@
+import os
+from typing import Tuple, cast
+
+import torch
+from model_def import MNistTrial
+
+
+class MNistFailable(MNistTrial):
+    def train_batch(self, batch, epoch_idx, batch_idx):
+        if "FAIL_AT_BATCH" in os.environ and int(os.environ["FAIL_AT_BATCH"]) == batch_idx:
+            raise Exception(f"failed at this batch {batch_idx}")
+
+        print("BATCH_IDX", batch_idx, "EPOCH IDX", epoch_idx)
+        batch = cast(Tuple[torch.Tensor, torch.Tensor], batch)
```

Comment:
why cast?

```suggestion
    def train_batch(self, batch: Tuple[torch.Tensor, torch.Tensor], epoch_idx, batch_idx):
        if "FAIL_AT_BATCH" in os.environ and int(os.environ["FAIL_AT_BATCH"]) == batch_idx:
            raise Exception(f"failed at this batch {batch_idx}")

        print("BATCH_IDX", batch_idx, "EPOCH IDX", epoch_idx)
```

Reply:
That is how the example is written? https://github.com/determined-ai/determined/blob/34c5bdb3f88a17cee0b4818fea9e9b4c10728fe6/examples/tutorials/mnist_pytorch/model_def.py#L85

I updated failable_model_def.py to just call the parent's class train_batch